src/App.tsx

```diff
@@ -8,7 +8,61 @@
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterOption } from './enums/filter-options';
+import { getFilteredTodos } from './components/helpers/getFilteredTodos';
+import { getTodosByQuery } from './components/helpers/getTodosByQuery';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [filterOption, setFilterOption] = useState<FilterOption>(
+    FilterOption.all,
+  );
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
```

Comment:
You are initializing your state with props. This approach can lead to bugs because React may not re-render the component when the props change. Consider deriving the state from props in the body of your component or using a useEffect hook to update the state when the props change.